src/arrayMethodJoin.js

```diff
@@ -5,7 +5,33 @@
  */
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
-    // write code here
+    let inFunctionSeparator = separator;
+
+    if (separator === null) {
+      inFunctionSeparator = 'null';
+    } else if (separator === '') {
```

Comment:
How does it differ from row 8? It does the same thing, you don't need this condition